### PR TITLE
M.Eng. team code submission

### PR DESCRIPTION
**Pull request by CTKnight**
- Merged our multi-map with latest augcog/master branch.

- Note that in the CMakeList, DLoopDetector is changed to be located by $ENV{DLoopDetector_INCLUDE_DIRS}. (This is written in the documentation/windows-buid-inst.md)

- The some urls to the dependencies like okvis and DLoopDetector in documentation/windows-buid-inst.md may be updated after our other PR to them are merged as well.

### CODE REVIEW

**Comment by adamchang2000** [comment]
I changed the CMakeLists.txt in CTKnight/OpenARK//submission, no longer comments out the SuiteSparse dependency to match with augcog/OpenARK/master

File: .gitignore [hunk 1]
```diff
@@ -288,6 +288,7 @@ CMakeFiles/
 cmake_modules/
 build/
 include/Version.h
+CmakeLists.txt
```
Comment (joemenke) [inline .gitignore]:
```suggestion

```

File: include/SparseMap.h [hunk 2]
```diff
@@ -66,13 +64,12 @@ class SparseMap {
   }
 
   void getMappedTrajectory(std::vector<int>& frameIdOut, std::vector<Eigen::Matrix4d>& trajOut){
-	frameIdOut.resize(frameMap_.size());
-    trajOut.resize(frameMap_.size());
-    size_t i=0;
+    //frameIdOut.resize(frameMap_.size());
```
Comment (joemenke) [inline include/SparseMap.h]:
```suggestion

```

File: include/SparseMap.h [hunk 3]
```diff
@@ -66,13 +64,12 @@ class SparseMap {
...
+    //trajOut.resize(frameMap_.size());
```
Comment (joemenke) [inline include/SparseMap.h]:
```suggestion

```

File: documentation/windows-buid-inst.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+1.    Download source from https://github.com/joemenke/DBoW2_Mod. Extract zip folder and cd to the extracted directory. (Note: this is a fork from the original git repository to make install on windows easier and add support for Brisk descriptors).
+
+2.    Follow CMAKE steps above.
+
+3.    You will need to add the environment variable `DBoW2_ROOT_DIR` and set it to the install location of DBoW2 (probably C:\Program Files\DBoW2) 
+
+Install DLoopDetector:
+
+1.    Download source from https://github.com/yohanmr/DLoopDetector. Extract zip folder and cd to the extracted directory. (Note: this is a fork from the original git repository with some minor modifications to support OpenARK integration).
```
Comment (joemenke) [inline documentation/windows-buid-inst.md]:
```suggestion
1.    Download source from https://github.com/joemenke/DLoopDetector. Extract zip folder and cd to the extracted directory. (Note: this is a fork from the original git repository with some minor modifications to support OpenARK integration).
```

File: documentation/windows-buid-inst.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    Make sure to select “add cmake to system PATH” either for current user or all users. 
+
+
+3. Install PCL1.8, easiest way is to use the all-in-one installer made by a kind PCL user: http://unanancyowen.com/en/pcl18/#Download
+
+    Make sure to select “add PCL to system PATH” either for current user or all users.
+
+    OpenNI2 will be installed as part of this installer. You will need to add the OpenNI2 dlls to the system path manually. The default location these are installed to is C:\Program Files\OpenNI2\Tools. See Step 4 in the opencv install instructions below for more details on modifying system variables. 
+
+4. **clone OpenARK repo from https://github.com/CTKnight/OpenARK/**
```
Comment (joemenke) [inline documentation/windows-buid-inst.md]:
```suggestion
4. **clone OpenARK repo from https://github.com/augcog/OpenARK/**
```

File: documentation/windows-buid-inst.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+1.    Download source from https://github.com/yohanmr/DLoopDetector. Extract zip folder and cd to the extracted directory. (Note: this is a fork from the original git repository with some minor modifications to support OpenARK integration).
+
+2.    Follow CMAKE steps above.
+
+3.    add enviromental variable `DLoopDetector_INCLUDE_DIRS` pointing to `${INSALL_DIR}/include/DLoopDetector`, typically it's `C:/Program Files/DLoopDetector/include/DLoopDetector`
+
+Install Okvis+:
+
+1.    Clone source from https://github.com/CTKnight/okvis/ and `git checkout add_reset`. Extract zip folder and cd to the extracted directory.
```
Comment (joemenke) [inline documentation/windows-buid-inst.md]:
```suggestion
1.    Clone source from https://github.com/joemenke/okvis/ and `git checkout add_reset`. Extract zip folder and cd to the extracted directory. 
```

**Review by joemenke — COMMENTED** [review]
Minor changes to clean for merge

**Pull request merged**